app/api/helpers/tasks.py

```diff
@@ -52,55 +52,56 @@
 
 
 @celery.task(name='send.email.post')
-def send_email_task(payload, headers):
-    message = Mail(from_email=payload['from'],
-                   to_emails=payload['to'],
-                   subject=payload['subject'],
-                   html_content=payload["html"])
-    if payload['attachments'] is not None:
-        for attachment in payload['attachments']:
-            with open(attachment, 'rb') as f:
-                file_data = f.read()
-                f.close()
-            encoded = base64.b64encode(file_data).decode()
-            attachment = Attachment()
-            attachment.file_content = FileContent(encoded)
-            attachment.file_type = FileType('application/pdf')
-            attachment.file_name = FileName(payload['to'])
-            attachment.disposition = Disposition('attachment')
-            message.add_attachment(attachment)
-    sendgrid_client = SendGridAPIClient(get_settings()['sendgrid_key'])
-    logging.info('Sending an email regarding {} on behalf of {}'.format(payload["subject"], payload["from"]))
-    try:
+def send_email_task(payload, headers, protocol, config=None):
+    if protocol == 'sendgrid':
```

Comment:
Why complicate things?

Create 2 functions, one for smtp, one for sendgrid

Reply:
Yeah, my original idea was to create a centralised function but this seems easier. Will update.